// primevue/tooltip/style/tooltipstyle.cjs.js
'use strict';

var BaseStyle = require('primevue/base/style');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var BaseStyle__default = /*#__PURE__*/_interopDefaultLegacy(BaseStyle);

var css = "\n@layer primevue {\n    .p-tooltip {\n        position:absolute;\n        display:none;\n        padding: .25em .5rem;\n        max-width: 12.5rem;\n    }\n\n    .p-tooltip.p-tooltip-right,\n    .p-tooltip.p-tooltip-left {\n        padding: 0 .25rem;\n    }\n\n    .p-tooltip.p-tooltip-top,\n    .p-tooltip.p-tooltip-bottom {\n        padding:.25em 0;\n    }\n\n    .p-tooltip .p-tooltip-text {\n        white-space: pre-line;\n        word-break: break-word;\n    }\n\n    .p-tooltip-arrow {\n        position: absolute;\n        width: 0;\n        height: 0;\n        border-color: transparent;\n        border-style: solid;\n    }\n\n    .p-tooltip-right .p-tooltip-arrow {\n        margin-top: -.25rem;\n        border-width: .25em .25em .25em 0;\n    }\n\n    .p-tooltip-left .p-tooltip-arrow {\n        margin-top: -.25rem;\n        border-width: .25em 0 .25em .25rem;\n    }\n\n    .p-tooltip.p-tooltip-top {\n        padding: .25em 0;\n    }\n\n    .p-tooltip-top .p-tooltip-arrow {\n        margin-left: -.25rem;\n        border-width: .25em .25em 0;\n    }\n\n    .p-tooltip-bottom .p-tooltip-arrow {\n        margin-left: -.25rem;\n        border-width: 0 .25em .25rem;\n    }\n}\n";
var classes = {
  root: 'p-tooltip p-component',
  arrow: 'p-tooltip-arrow',
  text: 'p-tooltip-text'
};
var TooltipStyle = BaseStyle__default["default"].extend({
  name: 'tooltip',
  css: css,
  classes: classes
});

module.exports = TooltipStyle;
